test/Dialect/FIRRTL/emit-metadata.mlir

```diff
@@ -0,0 +1,98 @@
+// RUN: circt-opt --pass-pipeline='firrtl.circuit(firrtl-emit-metadata)' %s | FileCheck %s
+
+firrtl.circuit "empty" {
+  firrtl.module @empty() {
+  }
+}
+// CHECK-LABEL: firrtl.circuit "empty"   {
+// CHECK-NEXT:    firrtl.module @empty() {
+// CHECK-NEXT:    }
+// CHECK-NEXT:  }
+
+//===----------------------------------------------------------------------===//
+// RetimeModules
+//===----------------------------------------------------------------------===//
+
+firrtl.circuit "retime0" attributes { annotations = [{
+    class = "sifive.enterprise.firrtl.RetimeModulesAnnotation",
+    filename = "./tmp/retime_modules.json"
+}]} {
+
+  firrtl.module @retime0() attributes { annotations = [{
+      class = "sifive.enterprise.firrtl.RetimeModuleAnnotation"
+  }]} { }
+
+  firrtl.module @retime1() { }
+
+  firrtl.module @retime2() attributes { annotations = [{
+      class = "sifive.enterprise.firrtl.RetimeModuleAnnotation"
+  }]} { }
+}
+// CHECK: firrtl.circuit "retime0"   {
+// CHECK:   firrtl.module @retime0() {
+// CHECK:   firrtl.module @retime1() {
+// CHECK:   firrtl.module @retime2() {
+// CHECK{LITERAL}: sv.verbatim "[\22{{0}}\22,\22{{1}}\22]"
+// CHECK-SAME:       output_file = #hw.output_file<"tmp/retime_modules.json", excludeFromFileList>
+// CHECK-SAME:       symbols = [@retime0, @retime2]
+
+//===----------------------------------------------------------------------===//
+// SitestBlackbox
+//===----------------------------------------------------------------------===//
+
+firrtl.circuit "Test" attributes { annotations = [{
+    class = "sifive.enterprise.firrtl.SitestBlackBoxAnnotation",
+    filename = "dut_blackboxes.json"
+  }]} {
+  firrtl.module @Test() attributes {annotations = [
+      {class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+  }
+}
+// CHECK:      firrtl.circuit "Test"   {
+// CHECK-NEXT:   firrtl.module @Test() attributes {annotations = [{class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+// CHECK-NEXT:   }
```

Comment:
Consider excluding this from the test as it's not pertinent to what's being tested.

Reply:
All the `CHECK-NEXT` actually helped me to verify that its not dropping an extra verbatim in there, since it does all the work of both annotations but only emits the verbatim op if the filename is empty

Reply:
If this exists only to check that no other `sv.verbatim` is in the test, then I'd suggest doing this as:

```
CHECK-LABEL: ...
CHECK:       sv.verbatim ...
CHECK-NOT:   sv.verbatim
```

Writing it the way is testing the lack of `sv.verbatim` via implication: "the strict structure of this test implies that no additional `sv.verbatim` exists".  I'd prefer if the test could more accurately state what it's trying to test other than the strict structure.  Also, I was wondering if this test was intentionally checking that the `MarkDUTAnnotation` was preserved.

This pedantry is coming from a more Scala-ish testing style of a sequence of tests that are all stating what they do:

```scala
foo should contain (bar)
foo.size should be (4)
```

I get antsy when I can't grok what the test is actually testing other than it produces a blob of text. 😄 

All that said, this is an extremely short test and it's _not_ bad to just keep it as is.  I'm just attempting to clarify where I'm coming from with "not pertinent".